### PR TITLE
Fix VS2015 build and refactored some minor things

### PR DESCRIPTION
**Pull request by ZehMatt**
I turned some sint32 result types to the bool macro/type as they only returned 0 or 1 and the name somewhat states that its boolean.

### CODE REVIEW

File: src/openrct2-ui/windows/Options.cpp [hunk 1]
```diff
@@ -1336,7 +1336,7 @@ static void window_options_dropdown(rct_window *w, rct_widgetindex widgetIndex,
         switch (widgetIndex) {
         case WIDX_CONSTRUCTION_MARKER_DROPDOWN:
             if (dropdownIndex != (gConfigGeneral.construction_marker_colour ? 1 : 0)) {
-                gConfigGeneral.construction_marker_colour = (uint8)dropdownIndex;
+                gConfigGeneral.construction_marker_colour = (uint8)dropdownIndex != 0;
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/Options.cpp]: What? Convert to bool and cast to uint8.

File: src/openrct2/windows/EditorObjectSelection.cpp [hunk 2]
```diff
@@ -897,8 +897,8 @@ void window_editor_object_selection_mousedown(rct_window *w, rct_widgetindex wid
         }
 
         if (!(gScreenFlags & SCREEN_FLAGS_TRACK_MANAGER)) {
-            dropdown_set_checked(DDIX_FILTER_SELECTED, _FILTER_SELECTED);
-            dropdown_set_checked(DDIX_FILTER_NONSELECTED, _FILTER_NONSELECTED);
+            dropdown_set_checked(DDIX_FILTER_SELECTED, _FILTER_SELECTED != 0);
+            dropdown_set_checked(DDIX_FILTER_NONSELECTED, _FILTER_NONSELECTED != 0);
```
Comment (duncanspumpkin) [inline src/openrct2/windows/EditorObjectSelection.cpp]: This does not look very nice at all. Surely we should just change the define to a bool value.

File: src/openrct2-ui/windows/Options.cpp (hunk 1 above)
Comment (IntelOrca) [inline src/openrct2-ui/windows/Options.cpp]: Remove `(uint8)`

Comment (IntelOrca) [inline src/openrct2-ui/windows/Options.cpp]: `(uint8)(dropdownIndex != 0)`

Comment (ZehMatt) [inline src/openrct2-ui/windows/Options.cpp]: Whats the cast about to begin with?

**Comment by ZehMatt** [comment]
If it happens to be that this will compile as C++ code one day isn't it logical to use the bool type.

**Comment by ZehMatt** [comment]
Since there is no plan for any further VS2015 support I'll just close it.

**Comment by janisozaur** [comment]
Until we move on to C++17, I think it's fine to submit compilation fixes for MSVC2015.

**Comment by ZehMatt** [comment]
Don't worry, you got it fixed already 👍

**Pull request closed without merging**